packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -462,4 +462,35 @@ export const useGridKeyboardNavigation = (
   useGridApiEventHandler(apiRef, 'columnHeaderKeyDown', handleColumnHeaderKeyDown);
   useGridApiEventHandler(apiRef, 'columnGroupHeaderKeyDown', handleColumnGroupHeaderKeyDown);
   useGridApiEventHandler(apiRef, 'cellKeyDown', handleCellKeyDown);
+
+  const keyDown = React.useCallback(
+    (event: any) => {
+      const focus = apiRef.current.state.focus;
+      if (focus.cell) {
+        const cell = focus.cell;
+        if (apiRef.current.getCellElement(cell.id, cell.field)) {
+          return;
+        }
+
+        apiRef.current.scrollToIndexes({
+          rowIndex: apiRef.current.getRowIndexRelativeToVisibleRows(cell.id),
+          colIndex: apiRef.current.getColumnIndex(cell.field),
+        });
+
+        apiRef.current.publishEvent(
+          'cellKeyDown',
+          apiRef.current.getCellParams(cell.id!, cell.field || ''),
+          event,
+        );
+      }
+    },
+    [apiRef],
+  );
+
+  React.useEffect(() => {
+    window.addEventListener('keydown', keyDown);
```

Comment:
when the focused cell unmounted body becomes the active element. I have added `scrollPostionChange` listener and check if the focused cell is unmounted then make `VirtualScrollerComponent` focus. is there a better way?

Reply:
hey @m4theushw 

I have been experimenting with different ideas and found a better way if we just store the active cell only in the DOM. it can be used for both rows and columns.

Reply:
it also works with `columnHeaders`